scripts/generating_yaml.py

```diff
@@ -0,0 +1,213 @@
+
+
+# In[2]:
+
+device_data = {'serial_into_USBhub_port' : [0, 1, 2, 3, 4, 5, 6, 7],
+               'camera_into_serial_port' : [1, 2, 3, 4],
+               'USBhub_SN' : ['OX518EFFE1', 'OXCD12637D'],
+               'camera_into_USBhub_port' : [0, 1, 2, 3, 4, 5, 6, 7],
+               'sbig_SN' : ['83F011167', '83F011791', '83F011639', '83F010801', '83F010774', '83F011758', '83F010771', '83F011810'],
+              'birger_SN': ['10858', '14287', '14286', '14285', '13281', '13134', '14284', '13208', '14276'],
+              'lens_SN': ['3360000099', '3360000063', '3360000087', '2850000067', '3150000110', '5370000054'],
+              'filter_ID': ['r2_1', 'r2_2', 'g2_3', 'g2_4', 'r2_5', 'r2_6', 'r2_7', 'g2_8', 'g2_9', 'ha1_10', 'ha1_11', 'g2_12', 'ha1_13', 'ha1_14']
+              }
+
+
+# In[3]:
+
+with open('serial_management.yaml', 'w') as file:
+    yaml.dump(device_data)
+    file.write(yaml.dump(device_data))
+
+with open('serial_management.yaml', 'r') as file:
+    data = yaml.load(file)
+
+
+# In[4]:
+
+def type_filename(type_filename_here):
+    return type_filename_here
+
+
+# In[5]:
+
+file_name = interactive(type_filename, type_filename_here="filename.yaml")
+
+
+# In[6]:
+
+data_list = {'name' : 'HuntsmanSSO',
+             'log_file' : 'huntsman',
+             'location' : {'name' : 'Siding Spring Observatory',
+                           'latitude' : -31.27, #degrees
+                           'longitude': 149.06, # Degrees
+                          'elevation': 1000.0, # Meters
+                           'utc_offset': 10.00, # Hours
+                           'horizon': 30, # Degrees
+                           'timezone': 'Australia/Sydney'
+                          },
+             'scheduler' : {'type' : 'dispatch',
+                           'fields_file' : 'targets.yaml'},
+             'dome' : {'template_dir' : 'resources/bisque'},
+             'guider' : {'template_dir' : 'resources/bisque/guider',
+                         'image_path': '/tmp/guide_image.fits',
+                        'bin_size' : 1},
+             'mount' : {'brand': 'bisque',
+                       'model' : 45,
+                       'driver' : 'bisque',
+                       'template_dir' : 'resources/bisque'},
+             'pointing' : {'threshold' : 0.05,
+                          'exp_time' : 30,
+                          'max_iterations' : 3},
+             'state_machine' : 'huntsman',
+             'directories' : {'base' : '/var/huntsman', 'images' : 'images',
+                              'webcam' : 'webcams', 'data' : 'data',
+                              'resources' : 'POCS/resources/',
+                              'targets' : '/var/huntsman/huntsman-pocs/conf_files',
+                              'mounts' : 'POCS/resources/mounts'
+             },
+    'cameras' : {'hdr_mode' : True,
+               'auto_detect' : False,
+                'devices' : [None]}}
+
+
+# In[7]:
+
+def a(birger_serial_number):
```

Comment:
Not really sure why the functions are defined as a,b,c etc. Is there a reason? Can be a bit confusing.

Reply:
they needed to be seperate functions so that the widgets could each have different labels. I just chose those letters as it was simple to copy and paste and change, but if it is confusing I can easily change them to something more logical

Reply:
Definitely name to something more logical.